app/components/button.jsx

```diff
@@ -0,0 +1,231 @@
+        style = {}, // may or may not be passed, Should be applied to the outer most tag
+        onDone = () => {}, // a function that is called when the button is clicked.  - if it exists
+        title = "", // text to display on hover
+        disabled = false, 
+        disableOnClick = false, // if true, the button gets disabled after click and stays disabled - prevents resubmission
+        children
+    } = props;
+
+    const [isDisabled, setIsDisabled] = useState(disabled);
+    const [parentIsHovered, setParentIsHovered] = useState(false);
+    // Stretch Goal 
+    // const [longPress, setLongPress] = useState(false);
+    // const [position, setPosition] = useState({ x: 0, y: 0 });
+    // const longPressTimer = useRef(null);
+
+    const classes = buttonStyles();
+    const combinedClassName = `${classes[className]} ${className}`;
+
+    // Stretch Goal 
+    // const startLongPress = () => {
+    //     const x = event.clientX || (event.touches && event.touches[0].clientX);
+    //     const y = event.clientY || (event.touches && event.touches[0].clientY);
+        
+    //     setPosition({ x, y });
+    //     longPressTimer.current = setTimeout(() => {
+    //         setLongPress(true);
+    //     }, 1000);
+    // }
+
+    // const stopLongPress = (event) => {
+    //     clearTimeout(longPressTimer.current);
+    //     if (longPress) {
+    //         setLongPress(false);
+    //     }
+    // }
+
+    return (
+        <div 
+            style={{ position: 'relative' }}
+            onMouseEnter={() => setParentIsHovered(true)}
+            onMouseLeave={() => setParentIsHovered(false)}
+        >
+            <button
+                className={`${combinedClassName} ${parentIsHovered ? 'hover' : ''}`}
+                style={style}
+                title={title}
+                disabled={isDisabled}
+                onClick={() => {
+                    if (onDone) onDone();
+                    if (disableOnClick) setIsDisabled(true);
+                }}
+
+                // onMouseDown={startLongPress}
+                // onMouseUp={stopLongPress}
+                // onMouseLeave={stopLongPress}
+                // onTouchStart={startLongPress}
+                // onTouchEnd={stopLongPress}
+            >
+                {children}
+            </button>
+            {/* Stretch Goal 
+            {longPress && 
+                <div style={{
+                    top: `${position.y}px`,
+                    left: `${position.x}px`,
+                    transform: 'translateX(-50%, -100%)',
+                    backgroundColor: 'white',
+                    border: '1px solid black',
+                    padding: '5px',
+                    borderRadius: '5px',
+                }}>
+                    {title}
+                </div>
+            } */}
+        </div>
+    )
+}
+
+function ModifierButton(props) {
+    return <Button {...props} className="modifierButton" />;
```

Comment:
The caller  of ModifierButton needs to be able to pass className to it, and any other props. 
One way to do this right is to have

```
...
    const {className, ...otherProps} = props;
    const classes=modifierButtonStyle();
    return <Button {...otherProps} className=cx(classes.modifierButton, className)
}